daemon/logger/fluentd/fluentd.go

```diff
@@ -0,0 +1,99 @@
+package fluentd
+
+import (
+	"io"
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/daemon/logger"
+	"github.com/fluent/fluent-logger-golang/fluent"
+)
+
+type Fluentd struct {
+	tag           string
+	containerID   string
+	containerName string
+	writer        *fluent.Fluent
+}
+
+const name = "fluentd"
+
+const defaultHostName = "localhost"
+const defaultPort = 24224
+const defaultTagPrefix = "docker"
+
+func init() {
+	if err := logger.RegisterLogDriver(name, New); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func New(ctx logger.Context) (logger.Logger, error) {
+	config := ctx.Config
+	port := defaultPort
+	host := defaultHostName
+	if config["port"] != "" {
+		portp, errp := strconv.Atoi(config["port"])
+		if errp != nil {
+			return nil, errp
+		}
+		port = portp
+	}
+	if config["host"] != "" {
+		host = config["host"]
+	}
+	tag := config["tag"]
```

Comment:
@tagomoris I said I'm just trying to have a pattern with other drivers but since we are prefixing log-opt with driver name each driver will handle its own opts as it wishes
BTW, this could be something like `--log-opt fluentd-tag=prefix.[id|full_id|name]` wdyt?